src/main/java/algorithms/curated170/medium/candycrush/CandyCrush.java

```diff
@@ -12,132 +10,79 @@ public static void main(String[] args) {
             {1, 2},
             {6, 2},
             {6, 2},
-            {1, 2},
-           
-        };
-        
-        candyCrush(board);
-   
-    }
-  
-     static boolean  signal = false; 
-      
-      public static int[][] candyCrush(int[][] board) {
-        int[][] boardCopy = marking(board);
-        
-        System.out.println("DROP:");
-        board =  drop(boardCopy);
-         
-         marking(board);
-         
-        return signal ? candyCrush(board) : board;
-
-        
-         
-    }
-     
-    public static int[][] marking(int[][] board){
-        signal = false;
-        int[][] boardCopy = new int[board.length][board[0].length];
+            {1, 2},};
 
-        for (int i = 0; i < board.length; i++) {
-            int low = 0;
+        CandyCrush candyCrush = new CandyCrush();
+        candyCrush.candyCrush(board);
 
-            for (int high = 1; high < board[i].length + 1; high++) {
-
-                while (high < board[i].length && board[i][low] != board[i][high] 
-                        || low < high && high == board[i].length) {
-
-                    if (board[i][low]!=0 && high - low  >= 3) {
-                        
-                        signal = true;
-                        while (low <high) {
-                            boardCopy[i][low] = 0;
-                            low++;
+    }
 
-                        }
-                    } else {
-                        while (low <high) {
+    public int[][] candyCrush(int[][] board) {
+        int[][] target = new int[board.length][board[0].length];
 
-                            boardCopy[i][low] = board[i][low];
-                            low++;
+        if (crush(board, target)) {
+            int[][] temp = board;
+            board = target;
+            target = temp;
+            return candyCrush(board);
```

Comment:
Why do we assign  values to temp and target here when we are not using them again?

Reply:
We can just input target to the method.